src/reflect/type.go

```diff
@@ -401,10 +422,49 @@ func (t Type) AssignableTo(u Type) bool {
 }
 
 func (t Type) Implements(u Type) bool {
-	if t.Kind() != Interface {
+	if u.Kind() != Interface {
 		panic("reflect: non-interface type passed to Type.Implements")
 	}
-	return u.AssignableTo(t)
+	if u.NumMethod() == 0 {
```

Comment:
This is subtly incorrect, as an interface can include unexported methods. I am not sure though whether there is an easy way to solve this right now. If you wish to leave this for further work, consider adding a comment about it.

Reply:
Relevant: https://github.com/golang/go/issues/22075

Reply:
Hmm this might be a bold thing to do but what if we reverted to the old (incorrect) behavior? It would make the implementation of `Implements` a lot easier, at least as long as we don't need to fully support it.

Reply:
By which you mean, the same behavior as Go currently has? I would agree with that, lets implement the same bugs :clown_face:

Reply:
Yes, the same buggy behavior (to be bug-compatible :wink:).

Reply:
Fixing this is slightly less trivial than I expected: I discovered that anonymous interface types (like `var x interface{ F() }`) do not have the correct `NumMethod`.

Reply:
What is the status with this one, @aykevl ?

Reply:
I need to look into this further, haven't done that yet.

Reply:
Reminder that we still need to resolve this. :smile_cat: